src/App.tsx

```diff
@@ -1,26 +1,224 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import {
+  USER_ID,
+  deleteTodo,
+  getTodos,
+  postTodo,
+  updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterTypes';
+import { filterTodos } from './utils/helpers';
+import { ErrorTypes } from './types/ErrorTypes';
+import { Header } from './Components/Header';
+import { Footer } from './Components/Footer';
+import { ErrorNotification } from './Components/ErrorNotification';
+import { TodoList } from './Components/TodoList';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.All);
+  const [error, setError] = useState<string>('');
+  const [value, setValue] = useState('');
+  const [inputDisabled, setInputDisabled] = useState(false);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    inputRef.current?.focus();
+  }, [todos, tempTodo]);
+
+  const handleError = (errorMessage: ErrorTypes) => {
+    setError(errorMessage);
+    setTimeout(() => {
+      setError('');
+    }, 3000);
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .then(() => {
+        if (loadingTodoIds.length > 0) {
+          setLoadingTodoIds([]);
+        }
+      })
+      .catch(() => {
+        handleError(ErrorTypes.OneMessage);
+      });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const removeTodo = (todoId: number) => {
+    setLoadingTodoIds(prev => [...prev, todoId]);
+    deleteTodo(todoId)
+      .then(() => {
+        setTodos(prevTodos => prevTodos.filter(todo => todo.id !== todoId));
+      })
+      .catch(() => {
+        // setLoadingTodoIds([]);
+        handleError(ErrorTypes.UnableToDelete);
+      })
+      .finally(() => {
+        setLoadingTodoIds(prev => prev.filter(item => item !== todoId));
+      });
+  };
+
+  const addTodo = (title: string) => {
+    const trimmedTitle = title.trim();
+
+    setTempTodo({
+      id: 0,
+      title,
+      completed: false,
+      userId: USER_ID,
+    });
+    setLoadingTodoIds(prev => [...prev, 0]);
+    postTodo(trimmedTitle)
+      .then(savedTodo => {
+        setTodos(prevTodos => [...prevTodos, savedTodo]);
+        setValue('');
+      })
+      .catch(() => {
+        handleError(ErrorTypes.UnableToAdd);
+      })
+      .finally(() => {
+        setInputDisabled(false);
+        setTempTodo(null);
+        setLoadingTodoIds([]);
```

Comment:
Remove only needed ids, the same as you have done in the previous function
`setLoadingTodoIds(prev => prev.filter(id=> id !== 0));`